### PR TITLE
Add DNS settings for community requests

### PR DESCRIPTION
**Pull request by davegson**
implements https://github.com/safing/portmaster/issues/246

### CODE REVIEW

File: portmaster/guides/dns-configuration.md [hunk 1]
```diff
@@ -123,6 +123,49 @@ dot://[2606:4700:4700::1113]:853?verify=cloudflare-dns.com&name=Cloudflare&block
 dot://[2606:4700:4700::1003]:853?verify=cloudflare-dns.com&name=Cloudflare&blockedif=zeroip
 ```
 
+### Community Suggested Server Settings
+
+Needs are different, that is why we list settings suggested by the community down below. Is something missing or out of date? [Make a report](https://github.com/safing/docs/issues/new?assignees=&labels=suggestion&template=suggest-feature.md) or [create a pull request](https://github.com/safing/docs)
```
Comment (dhaavi) [inline portmaster/guides/dns-configuration.md:128]: Missing dot at the end.

File: portmaster/guides/dns-configuration.md [hunk 2]
```diff
@@ -123,6 +123,49 @@ dot://[2606:4700:4700::1113]:853?verify=cloudflare-dns.com&name=Cloudflare&block
...
+
+#### BlahDNS [](https://blahdns.com/)
+
+```
+# Malware Protection, Ad Blocking
+🇨🇭 dot://45.91.92.121:853?verify=dot-ch.blahdns.com&name=BlahDNSch&blockedif=empty
+🇯🇵 dot://139.162.112.47:853?verify=dot-jp.blahdns.com&name=BlahDNSjp&blockedif=empty
+🇸🇬 dot://192.53.175.149:853?verify=dot-sg.blahdns.com&name=BlahDNSsg&blockedif=empty
+🇩🇪 dot://78.46.244.143:853?verify=dot-de.blahdns.com&name=BlahDNSde&blockedif=empty
+🇫🇮 dot://95.216.212.177:853?verify=dot-fi.blahdns.com&name=BlahDNSfi&blockedif=empty
```
Comment (dhaavi) [inline portmaster/guides/dns-configuration.md:138]: Flags are nice, but no one has usually memorized them all.

While you'd normally only need your own or near to you country's flag, maybe adding the name somehow might be nice. Can we add an alt tag to the flag?

File: portmaster/guides/dns-configuration.md [hunk 3]
```diff
@@ -123,6 +123,49 @@ dot://[2606:4700:4700::1113]:853?verify=cloudflare-dns.com&name=Cloudflare&block
...
+# Malware Protection, Ad & Tracker Blocking
+dot://116.202.176.26:854?verify=dot.libredns.gr&name=LibreDNS&blockedif=zeroip
+```
+
+#### NextDNS [](https://www.nextdns.io/)
+
+```
+# Malware Protection, Ad Blocking
+dot://45.90.28.108:853?verify=b4ba5f.dns.nextdns.io&name=NextDNS&blockedif=empty
+dot://45.90.30.108:853?verify=b4ba5f.dns.nextdns.io&name=NextDNS&blockedif=empty
```
Comment (dhaavi) [inline portmaster/guides/dns-configuration.md:156]: Everyone has to add their own username here! So we cant just do it like this.

I would more or less copy/paste the instructions from the GH issue.

File: portmaster/guides/dns-configuration.md [hunk 4]
```diff
@@ -123,6 +123,49 @@ dot://[2606:4700:4700::1113]:853?verify=cloudflare-dns.com&name=Cloudflare&block
...
+🇸🇬 dot://192.53.175.149:853?verify=dot-sg.blahdns.com&name=BlahDNSsg&blockedif=empty
+🇩🇪 dot://78.46.244.143:853?verify=dot-de.blahdns.com&name=BlahDNSde&blockedif=empty
+🇫🇮 dot://95.216.212.177:853?verify=dot-fi.blahdns.com&name=BlahDNSfi&blockedif=empty
+```
+
+#### LibreDNS [](https://libredns.gr/)
+
+```
+# Malware Protection
+dot://116.202.176.26:853?verify=dot.libredns.gr&name=LibreDNS&blockedif=empty
```
Comment (dhaavi) [inline portmaster/guides/dns-configuration.md:145]: I would bet they also use a zero IP for blocking here too.

File: portmaster/guides/dns-configuration.md (hunk 2 above)
Comment (dhaavi) [reply portmaster/guides/dns-configuration.md:138]: From my tests they use block with `zeroip`.

**Review by dhaavi — CHANGES_REQUESTED** [review]
Very nice!

- A couple minor issues.
- NextDNS needs a bit more fixing.

Also, the port 853 is the default for DoT servers, so you may omit it.
See https://github.com/safing/portmaster/blob/82a7350f9afee91f9948912857bbf943969f9b81/resolver/resolvers.go#L108-L109

**Comment by davegson** [comment]
thanks, implemented everything as suggested

**Pull request merged**